lazy-imports-import.html

```diff
@@ -22,23 +23,26 @@
       var imports = fromElement.querySelectorAll(query);
       var promises = [];
       for (var i = 0; i < imports.length; i++) {
-        promises.push(new Promise(function(resolve) {
-          var href = imports[i].getAttribute('href');
-          var onLoad = function() {
-            resolve({loaded: href});
-          };
-          var onFailure = function() {
-            resolve({failed: href});
-          };
-          var resolved = callingElement.resolveUrl(href);
-          if (!__loadedImports[resolved]) {
-            __loadedImports[resolved] = true;
+        var href = imports[i].getAttribute('href');
+        var resolvedUrl = callingElement.resolveUrl(href);
+
+        if (__failedImports[resolvedUrl] || !__loadedImports[resolvedUrl]) {
```

Comment:
I think this would be a bit simpler if instead of storing state for the success and failed cases, you just stored the Promise and return it if it exists. The Promise will either be pending, resolved, or rejected, which takes care of the different cases.

So:

```ts
if (__importPromises.hasOwnProperty(resolvedUrl)) {
  return __importPromises[resolvedUrl];
}
__importPromises[resolvedUrl] = new Promise(...);
```

The onLoad and onFailure handlers will be simpler too.